core/trino-main/src/main/java/io/trino/operator/TaskContext.java

```diff
@@ -349,6 +353,17 @@ public long getPhysicalWrittenDataSize()
         return physicalWrittenBytes;
     }
 
+    public void setMaxWriterCount(int maxWriterCount)
+    {
+        this.maxWriterCount.compareAndSet(0, maxWriterCount);
```

Comment:
use `getAndSet` and make sure previous value is -1 (or the same as new `maxWriterCount` if we expect `setWriterount` to be called multiple times)